diff -Nru gri-2.12.1.old/popt/poptconfig.c gri-2.12.1/popt/poptconfig.c
--- gri-2.12.1.old/popt/poptconfig.c	Wed Sep 25 16:39:51 2002
+++ gri-2.12.1/popt/poptconfig.c	Sat Mar 29 17:21:59 2003
@@ -129,7 +129,7 @@
 }
 
 int poptReadDefaultConfig(poptContext con, int useEnv) {
-    char * fn, * home;
+    char * fn, * home, * etc_dir;
     int rc;
 
     if (!con->appName) return 0;
@@ -138,13 +138,21 @@
     if (rc) return rc;
     if (getuid() != geteuid()) return 0;
 
-    if ((home = getenv("HOME"))) {
-	fn = alloca(strlen(home) + 20);
-	strcpy(fn, home);
-	strcat(fn, "/.popt");
-	rc = poptReadConfigFile(con, fn);
-	if (rc) return rc;
-    }
+	if ((etc_dir = getenv("ETC_DIR"))) {
+		fn = alloca(strlen(etc_dir) + 20);
+		strcpy(fn, etc_dir);
+		strcat(fn, "/popt");
+		rc = poptReadConfigFile(con, fn);
+		if (rc) return rc;
+	} else {
+    	if ((home = getenv("HOME"))) {
+		fn = alloca(strlen(home) + 20);
+		strcpy(fn, home);
+		strcat(fn, "/.popt");
+		rc = poptReadConfigFile(con, fn);
+		if (rc) return rc;
+    	}
+	}
 
     return 0;
 }
diff -Nru gri-2.12.1.old/startup.cc gri-2.12.1/startup.cc
--- gri-2.12.1.old/startup.cc	Wed Sep 25 16:39:51 2002
+++ gri-2.12.1/startup.cc	Sat Mar 29 17:27:06 2003
@@ -1137,12 +1137,17 @@
 	_done = 0;
 	_margin.assign("| ");
 #if defined(HAVE_GETENV)
-	char *            home = egetenv("HOME");
-	if (!home) {
-		warning("Cannot getenv(HOME), so cannot perform grirc file");
-		return;
+	char * etc_dir = egetenv("ETC_DIR");
+	if (etc_dir) {
+		sprintf(grircname, "%s/%s", etc_dir, GRIRC_FILE+1);
+	} else {
+		char *            home = egetenv("HOME");
+		if (!home) {
+			warning("Cannot getenv(HOME), so cannot perform grirc file");
+			return;
+		}
+		sprintf(grircname, "%s/%s", home, GRIRC_FILE);
 	}
-	sprintf(grircname, "%s/%s", home, GRIRC_FILE);
 #else
 	strcpy(grircname, GRIRC_FILE);
 #endif
